enterprise/functions/src/main/java/io/crate/window/NthValueFunction.java

```diff
@@ -52,13 +53,22 @@ public FunctionInfo info() {
     }
 
     @Override
-    public Object execute(int rowIdx, WindowFrameState currentFrame, List<? extends CollectExpression<Row, ?>> expressions, Input... args) {
+    public Object execute(int rowIdx,
+                          WindowFrameState currentFrame,
+                          List<? extends CollectExpression<Row, ?>> expressions,
+                          Input... args) {
         if (isNewFrame(seenFrameUpperBound, currentFrame)) {
             seenFrameUpperBound = currentFrame.upperBoundExclusive();
-            int frameSize = currentFrame.size();
-            Row lastRowInFrame = new RowN(currentFrame.getRowAtIndexOrNull(frameSize - 1));
+            int zeroBasedIndex = ((Number) args[1].value()).intValue() - 1;
+            Object[] nthRowCells = currentFrame.getRowAtIndexOrNull(zeroBasedIndex);
+            if (nthRowCells == null) {
+                resultForCurrentFrame = null;
+                return null;
```

Comment:
`return  resultForCurrentFrame`  ?

Or if you prefer add an else branch and keep a single return at the end. 
My personal preference is to return early though as it makes the code more readable imho

Reply:
I don't think branching the logic is more readable than an early return statement. I believe we generally implement early return across the codebase. If we want to change this because we think it's more readable to only have one return statement we should look at changing the codestyle rules, but I personally don't think we should enforce anything like that.

Reply:
> I don't think branching the logic is more readable than an early return statement. 

I completely agree I'm also in favor on returning early (that's what I said!)

so should we change `return null;` -> `return resultForCurrentFrame;`